sbol2/SBOL2Serialize.py

```diff
@@ -33,41 +33,37 @@
 rdfNS = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"
 sbolNS = "http://sbols.org/v2#"
 
+# This is a module global that is used to register ownership relationships
+# between classes. The serializer uses this to generate structured XML from
+# flat RDF/XML
+OWNERSHIP_PREDICATES = {}
+
 
 def is_ownership_relation(g, triple):
-    subject = triple[0].toPython()
-    predicate = triple[1].toPython()
+    subject = triple[0]
+    predicate = triple[1]
     obj = triple[2]
 
-    ownership_predicates = {
-        sbolNS + 'module',
-        sbolNS + 'mapsTo',
-        sbolNS + 'interaction',
-        sbolNS + 'participation',
-        sbolNS + 'functionalComponent',
-        sbolNS + 'sequenceConstraint',
-        sbolNS + 'location',
-        sbolNS + 'sourceLocation',
-        sbolNS + 'sequenceAnnotation',
-        sbolNS + 'measure'
-    }
-
-    if predicate in ownership_predicates:
-        return True
+    if predicate not in OWNERSHIP_PREDICATES:
+        return False
 
     # SBOL2 reuses the "component" predicate as both an ownership predicate (in
     # the case of ComponentDefinition) and a referencing one (in the case of
-    # SequenceAnnotation).
-    #
-    if predicate == sbolNS + 'component':
-        if (triple[0], RDF.type, URIRef(sbolNS + 'SequenceAnnotation')) in g:
-            return False
-        else:
-            return True
-
+    # SequenceAnnotation). This case requires we check the OWNERSHIP_PREDICATES
+    # register
+    if (subject, RDF.type, OWNERSHIP_PREDICATES[predicate]) in g:
+        return True
     return False
 
 
+def register_ownership_relation(parent_type, predicate):
+    # Encapsulates and populates the OWNERSHIP_PREDICATES dictionary.
+    #
+    # :param parent_type: The RDF type of the parent class
+    # :param predicate: The URI for the property.
+    OWNERSHIP_PREDICATES[URIRef(predicate)] = URIRef(parent_type)
```

Comment:
I'm worried about this. It means that a given predicate can be an ownership relation in the context of only one parent type. Is that guaranteed to be true? It feels like this could cause problems if a predicate is an ownership relation for more than one parent type.

I had originally suggested that `OWNERSHIP_PREDICATES` has the parent_type as its keys, and the values would be lists of predicates. We could still do that and modify the checks in `is_ownership_relation` accordingly. I think this is a small departure from where we are, and provides better flexibility regarding ownership relationships.

Reply:
I updated the branch to allow multiple parent types for a given predicate.